cmd/lukso/validator.go

```diff
@@ -48,174 +47,223 @@ type DepositDataKey struct {
 	DepositCliVersion     string   `json:"deposit_cli_version"`
 }
 
-func sendDeposit(ctx *cli.Context) error {
-	// Dialing clients and creating bindings
+type depositController struct {
+	c context.Context
+
+	eth        *ethclient.Client
+	genDep     *bindings.LYXe
+	dep        *bindings.EthereumDeposit
+	senderAddr common.Address
+
+	depositKeys   []DepositDataKey
+	startingIndex int
+	keysNum       int
+	txOpts        *bind.TransactOpts
+}
+
+func newDepositController(rpc string, depositKeys []DepositDataKey, startingIndex int) (dc depositController, err error) {
+	c := context.Background()
+	keysLen := len(depositKeys)
+	if keysLen < 1 {
+		err = errDepositNotProvided
+
+		return
+	}
+
 	log.Info("Dialing up blockchain for gas info...")
-	eth, err := ethclient.Dial(ctx.String(rpcFlag))
+	eth, err := ethclient.Dial(rpc)
 	if err != nil {
-		return err
+		return
 	}
 
-	lyxMock, err := bindings.NewLYXe(common.HexToAddress(lyxeContractAddress), eth)
+	genDep, err := bindings.NewLYXe(common.HexToAddress(lyxeContractAddress), eth)
 	if err != nil {
-		return err
+		return
 	}
 
-	ethDeposit, err := bindings.NewEthereumDeposit(common.HexToAddress(depositContractAddress), eth)
+	dep, err := bindings.NewEthereumDeposit(common.HexToAddress(depositContractAddress), eth)
 	if err != nil {
-		return err
+		return
 	}
 
-	c := context.Background()
+	if startingIndex < 0 {
+		log.Error("Couldn't send deposits: starting index is smaller than 0")
+		err = errIndexOutOfBounds
 
-	var supplyAmount int
-
-	depositPath := ctx.String(depositDataJson)
-	isGenesisDeposit := ctx.Bool(genesisDepositFlag)
-
-	if isGenesisDeposit {
-		supplyAmount, err = chooseSupply()
-		if err != nil {
-			return err
-		}
+		return
 	}
+	if startingIndex >= keysLen {
```

Comment:
Are you sure that this shouldn't look like `startingIndex > keysLen`? Because you've implemented like `keysLen < 1` so it means it's no `0-indexed`.